### ISSUE TITLE
Istio deployment template metadata labels should have version

### ISSUE THREAD

**Issue by jmazzitelli** [istio/istio#8413]
**Describe the bug**

About a month ago, I submitted an issue https://github.com/istio/istio/issues/7639 asking the labels that Istio puts on its own components match Istio's own recommendations for user components (specifically the "app" and "version" label is what I cared about).

That issue was implemented in PR https://github.com/istio/istio/pull/7685 but it is not fully complete. Some Istio deployments do not put "version" label in their template metadata - so the Istio pods do not get a "version" and thus the Istio telemetry that is collected and stored in Prometheus assign the metric data with a version label of "unknown".

For example, see this:

https://github.com/istio/istio/pull/7685/files#diff-4f6a75809d8436568b4338b4bf0f83b9R208

This is where the istio-policy and istio-telemetry deployments are defined. Notice that the deployment itself has app/version labels but the pods created by those deployments will not have a version label, only an app label (see the template/metadata/labels section - it is missing version label).

Now when Istio collects telemetry for its own mixer components (in this case policy and telemetry pods) it won't know what version to label the metric data. For example, I'm running Istio 1.1, and in Prometheus, here is the metric timeseries I see for the bookinfo's productpage requests into istio-policy:

Prometheus query: `istio_requests_total{destination_workload_namespace="istio-system", destination_workload="istio-policy", source_app="productpage"}`

```
istio_requests_total{
connection_security_policy="none",
destination_app="policy",
destination_service="istio-policy.istio-system.svc.cluster.local",
destination_service_name="istio-policy",
destination_service_namespace="istio-system",
destination_version="unknown", <<---- LOOK HERE! unknown version!
destination_workload="istio-policy",
destination_workload_namespace="istio-system",
destination_principal="unknown",
instance="172.17.0.14:42422",
job="istio-mesh",
reporter="destination",
request_protocol="grpc",
response_code="200",
source_app="productpage",
source_principal="unknown",
source_version="v1",
source_workload="productpage-v1",
source_workload_namespace="bookinfo"}
```
**Expected behavior**

All Istio Deployment components, when installed, will give the pods they create/manage proper app and version labels - this is to ensure metric collection for those pods can be labeled properly. In the case of version labels (which look to be the ones missing) the deployment should simply give the pods a version label whose value is the same value as the version label that is on the deployment itself.

**Steps to reproduce the bug**

Install Istio 1.1, look at the Istio deployments in the istio-system namespace - look at the template/metadata/labels for the deployments and noticed some (all?) are missing a version label.

**Version**

```
$ istiooc version
oc v3.10.0+maistra-0.1.0-3+93134ea
kubernetes v1.10.0+b81c8f8
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://192.168.1.19:8443
openshift v3.10.0+8e19bad-29
kubernetes v1.10.0+b81c8f8
```

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:17:39Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.0+b81c8f8", GitCommit:"b81c8f8", GitTreeState:"clean", BuildDate:"2018-08-30T19:54:31Z", GoVersion:"go1.9.7", Compiler:"gc", Platform:"linux/amd64"}
```

**Is Istio Auth enabled or not?**

No.

**Environment**

OS 3.10

**Comment by jaredeis** [comment]
This would make testing of new istio components in a single namespace easier as you can point the Service selector to the new version of the component (pilot, telemetry, policy, citadel, etc) to the new version and fail back more quickly if there's a breaking change.

**Comment by jmazzitelli** [comment]
@ymesika since you implemented PR #7685 perhaps you would be the best to look at this issue?
